### PR TITLE
DAOS-12917 control: Add hugepages for extra system XStream

### PR DESCRIPTION
**Pull request by tanabarr**
For MD-on-SSD we need extra hugepage memory to serve an extra system
XStream to run RDB, for now treat this as an extra target when
calculating the number of huge pages required to satisfy the DMA
buffer requirements but in reality the extra needed might not be a
full GiB.

Required-githooks: true

### Before requesting gatekeeper:

* [x] Two review approvals and any prior change requests have been resolved.
* [x] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [x] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [x] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [x] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

File: src/control/lib/control/auto.go [hunk 1]
```diff
@@ -1142,24 +1141,20 @@ func getThreadCounts(log logging.Logger, ec *engine.Config, coresPerEngine int)
...
-	if err != nil {
-		return nil, errors.Wrap(err, "unable to calculate minimum hugepages")
 	}
 
 	cfg := config.DefaultServer().
 		WithAccessPoints(accessPoints...).
 		WithFabricProvider(ecs[0].Fabric.Provider).
 		WithEngines(ecs...).
-		WithControlLogFile(defaultControlLogFile).
-		WithNrHugePages(reqHugePages)
```
[the remark this answers is not in the document]
Comment (tanabarr) [reply src/control/lib/control/auto.go:1162]: It feels better now the calculation is only performed in one place.

**Review by knard-intel — DISMISSED** [review]
LGTM

File: ci/bandit.config [hunk 2]
```diff
@@ -30,7 +30,6 @@
 # B306 : mktemp_q
 # B307 : eval
 # B308 : mark_safe
-# B309 : httpsconnection
```
Comment (kjacque) [inline ci/bandit.config:33]: What's the reason for the changes in this file?

File: src/control/cmd/daos_server/auto_test.go [hunk 3]
```diff
@@ -361,7 +361,7 @@ func TestDaosServer_Auto_confGen(t *testing.T) {
 				},
 			},
 			expCfg: control.MockServerCfg("ofi+psm2", tmpfsEngineCfgs).
-				WithNrHugePages(18432).
+				WithNrHugePages(19456).
```
Comment (kjacque) [inline src/control/cmd/daos_server/auto_test.go:364]: Magic numbers... is there a formula the test could use to calculate this?

File: src/control/cmd/dmg/auto_test.go [hunk 4]
```diff
@@ -244,7 +244,7 @@ func TestAuto_confGen(t *testing.T) {
 				{storHostResp},
 			},
 			expCfg: control.MockServerCfg("ofi+psm2", tmpfsEngineCfgs).
-				WithNrHugePages(16384).
+				WithNrHugePages(17408).
```
Comment (kjacque) [inline src/control/cmd/dmg/auto_test.go:247]: Same comment as above

File: ci/bandit.config (hunk 2 above)
Comment (tanabarr) [reply ci/bandit.config:33]: The bandit fix (d64db3f414a9f6f2a89c0c2e4f9da8add6ddb1de) that was landed to master was needed in order to get CI tests to run, this has not yet been merged into the feature branch so I merged the commit into this PR.

File: src/control/cmd/daos_server/auto_test.go (hunk 3 above)
Comment (tanabarr) [reply src/control/cmd/daos_server/auto_test.go]: done

**Review by knard-intel — APPROVED** [review]

File: src/control/cmd/daos_server/auto_test.go [hunk 5]
```diff
@@ -237,7 +237,8 @@ func TestDaosServer_Auto_confGen(t *testing.T) {
 				},
 			},
 			expCfg: control.MockServerCfg("ofi+psm2", exmplEngineCfgs).
-				WithNrHugePages(18432).
+				// 18 targets * 2 engines * 512 pages
+				WithNrHugePages(18 * 2 * 512).
```
Comment (kjacque) [inline src/control/cmd/daos_server/auto_test.go:241]: Definitely helps make it clearer, thanks. IMO creating named constants would be even better, removing the need for the comments.

Comment (tanabarr) [reply src/control/cmd/daos_server/auto_test.go:241]: Is the change made not satisfactory? Do I need to make any further changes?

**Comment by tanabarr** [comment]
CI run failed with 2 unrelated but identified issues:
https://daosio.atlassian.net/browse/DAOS-12857
https://daosio.atlassian.net/browse/DAOS-12922

**Review by kjacque — APPROVED** [review]
